### DISCUSSION TITLE
Clarification on redirect port in HTTPRequestRedirectFilter

### DISCUSSION THREAD

**Discussion by arkodg** [kubernetes-sigs/gateway-api#1725] [Q&A]
The [HTTPRequestRedirectFilter ](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPRequestRedirectFilter) docs describe the `port` field as 
```
Port is the port to be used in the value of the Location header in the response. When empty, port (if specified) of the request is used.
```

Hoping to get more clarification on `When empty, port (if specified) of the request is used.` 
* Is the `port of the request` referring to the Layer 4 port the request is to set or
* The port section of host within the Host header of the HTTP request

**Comment by arkodg** [comment]
Envoy based implementations such as Contour ( @skriss @sunjayBhatia ), Emissary (@AliceProxy), EnvoyGateway and Istio (@howardjohn ) might be affected by this definition.
here's the envoy issue https://github.com/envoyproxy/envoy/issues/17318

**Comment by howardjohn** [comment]
Complex example we should make sure we can answer:

TCP LB on port 1000 -> Gateway listener 2000 (backed by a LB Service with port 2000 target port 3000)

User sends `curl tcplb:1000`. The proxy sees a request with l4 port 3000, which it knows is for Gateway.Listener port 2000, and `Host: tcplb:1000`.

So I think the 3 cases are:
1. Host header
2. Gateway Listener
3. actual L4 layer

Istio does (2)

**Reply by robscott** [reply]
When I was first reading this, I was expecting 1) to be the interpretation of the spec since this is a concept defined in HTTPRoute. With that said, I can see the reasoning behind attaching this to Gateway Listener port. That seems like it would be problematic for GAMMA though?

Each possible approach has limitations, and those are best solved by asking users to specify port. Whatever approach we choose, we should probably update our docs to:

1. Explain the limitations of inferring ports in redirects with Envoy + what implementations will be looking at (maybe host header)
2. Recommend setting port in redirects to ensure the most predictable behavior

**Reply by arkodg** [reply]
This line in the definition `port (if specified) of the request is used` makes me interpret it as 1.
